### ISSUE TITLE
Hide DS4 never works

### ISSUE THREAD

**Issue by mrvice** [Ryochan7/DS4Windows#1240]
I only got it to work once with the jayking version.
sadly it´s also not working with this version you provide
regardless what i do controller is always visible in windows game controller settings.
and i always get the error with exclusive mode.
I litterly did closed all programs nothing was running still not working very annoying.
even starting DS4 as admin did not help
----
iam afraid of using this hidguardian thing i personaly don´t like programs that can execute on kernel lvl not sure how safe this thing is. (specialy afraid for anti cheat tools)

**Comment by mika-n** [comment]
This really is PC environment specific thing. You have to try to find which app or background service is hijacking the DS4 gamepad and this way preventing "exclusive mode" (=Hide DS4 Controller) opton to work. We have no way to know which apps you are using. The following page names few common apps and gives a tip how to try to find an app process using a device handle.
https://github.com/Ryochan7/DS4Windows/wiki/Exclusive-Mode-(Hide-DS4-Controller-config-option)-tips-and-issues

If you cannot find the app hijacking DS4 gamepad then unfortunately HidGuardian is the next best option to hide physical DS4 gamepad (in some cases the only option).

**Comment by Ryusennin** [comment]
You are overly paranoid. From my personal experience, HidGuardian poses no 'legal' issue with strict online games like Destiny 2 or The Division 1/2.

**Comment by Ryochan7** [comment]
Please don't write a wall of bold text again.

The **Hide DS4 Controller** technique does not really work with a lot of people's configurations anymore. It does not help that even the Windows Store app can cause problems if you have the DS4 plugged in at startup. In most cases, some Nvidia service is probably running in the background and that is likely the culprit; just closing the GUI for some programs is not enough. At least the current iteration works in more cases than the Jays2Kings version which would only work on 32-bit systems.

HidGuardian is the only real way around this problem at the system level; the only better way would be a dedicated DS4 driver. It is too bad that several compromises have to be made to utilize HidGuardian especially the vanilla 1.14.3.0 version. It also does not help that the project is defunct for the most part. I have only been actively using the driver for the past few months. I had to customize it further to make it behave how I want.

No game should be looking for HidGuardian and it is a properly signed driver. There should be no problem with it being installed. If the concern is that great then maybe you should not be playing that game in the first place. Also, the HidGuardian source code is fairly small so it is easy to inspect what the driver does if you can read C code.

TL;DR: This is an unsolved problem.

**Comment by lmills87** [comment]
Hey Ryochan7, just wanted to make a statement and ask a question.

Statement: For me it seems that disabling the Nvidia Shadowplay overlay (via the GeForce Experience settings menu) and then closing/re-opening DS4Windows fixes the issue (at least for the application I'm testing it with, which is RetroArch). I can even enable the Shadowplay overlay after doing this procedure and the Hide DS4 Controller option will continue to function properly from this point on. I saw a similar thread on the Jays2Kings (spelling?) fork and this was the suggested fix, which seemed to work for several people.

Question 1: Is there anything you can potentially do with that information to create a workaround for those of us this procedure solves the problem for?

Question 2: Otherwise, are you suggesting that HidGuardian is the best way to accomplish the same thing? Are you saying that HidGuardian is defunct at this point, and if so, is that a bad thing or a reason that we might not want to use it?

**Comment by Ryochan7** [comment]
I think I have seen some batch scripts floating around when Sekiro was released. That is a very temporary workaround that would have to be done every time a DS4 device is re-connected. Also, other programs besides the GeForce Experience suite can cause the same type of issue to surface.

HidGuardian is the more sure way to work around the exclusive mode problem. Although, I don't use the vanilla version but my own modified version. My modified version only handles the initial connection rather than trying to forward all HID requests up the stack. That change eliminates the chance for added input lag so native input performance can be achieved. My modified version of HidGuardian is only available in source form.

https://github.com/Ryochan7/HidGuardian/tree/gottagofast

At least for now, Nefarius has discouraged people from using HidGuardian unless absolutely necessary and he is more focused on other projects so he has labeled HidGuardian defunct. Problems regarding Windows Updates probably played a factor on his decision as well. From his own experiments, it would seem like going with the user mode driver approach would be the future for blocking devices from system access.

I have only actively started using HidGuardian fairly recently and I keep finding more uses for it. Besides DS4Windows, I have used HidGuardian with Touchmote and my most recent Switch Pro controller project. I had to come up with my own tools to work with HidGuardian due to issue I have had with existing tools.
